cpp/include/cudf/detail/aggregation/aggregation.hpp

```diff
@@ -581,7 +581,7 @@ class collect_list_aggregation final : public rolling_aggregation {
 /**
  * @brief Derived aggregation class for specifying COLLECT_SET aggregation
  */
-class collect_set_aggregation final : public aggregation {
+class collect_set_aggregation final : public rolling_aggregation {
```

Comment:
Why do we inherit collect set from `rolling_aggregation` instead of `aggregation`? We need it for both rolling window and groupby, don't we?

Reply:
Because `rolling_aggregation` is virtually inherited from `aggregation`.

Reply:
Maybe both are wrong, as `collect_set_aggregation` and `collect_set_aggregation` are used not only in rolling window but in groupby. Can you change to:
```
class collect_list_aggregation final : public aggregation 
...
class collect_set_aggregation final : public aggregation 
```
and test if they can compile and unit tests all pass please?

Reply:
Hi @ttnghia, I tried on replacing `rolling_aggregation` with `aggregation`. And I got compiling error:
```
/home/alfredxu/workspace/codes/cudf/cpp/src/aggregation/aggregation.cpp:454:74: error: could not convert ‘std::make_unique(_Args&& ...) [with _Tp = cudf::detail::collect_list_aggregation; _Args = {cudf::null_policy&}; typename std::_MakeUniq<_Tp>::__single_object = std::unique_ptr<cudf::detail::collect_list_aggregation, std::default_delete<cudf::detail::collect_list_aggregation> >]()’ from ‘unique_ptr<cudf::detail::collect_list_aggregation,default_delete<cudf::detail::collect_list_aggregation>>’ to ‘unique_ptr<cudf::rolling_aggregation,default_delete<cudf::rolling_aggregation>>’
```

Reply:
```cpp
/// Factory to create a COLLECT_LIST aggregation
template <typename Base = aggregation>
std::unique_ptr<Base> make_collect_list_aggregation(null_policy null_handling)
{
  return std::make_unique<detail::collect_list_aggregation>(null_handling);
}
template std::unique_ptr<aggregation> make_collect_list_aggregation<aggregation>(
  null_policy null_handling);
template std::unique_ptr<rolling_aggregation> make_collect_list_aggregation<rolling_aggregation>(
  null_policy null_handling);
```

I think it is because we can not return `std::unique_ptr<rolling_aggregation>` after we made the replacement.

Reply:
I see. Thanks 😄